repose-aggregator/components/services/open-tracing-service/open-tracing-service-api/src/main/resources/META-INF/schema/config/open-tracing.xsd

```diff
@@ -0,0 +1,313 @@
+        </xs:complexContent>
+    </xs:complexType>
+
+
+    <xs:complexType name="JaegerConnection" abstract="true">
+        <xs:attribute name="host" type="xs:string">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        Tracer host.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+
+        <xs:attribute name="port" type="PositiveInt">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        Tracer port.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="JaegerConnectionHttp">
+        <xs:complexContent>
+            <xs:extension base="JaegerConnection">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            Indicates that a http connection is to be used to connect to the tracer.
+                            Username/password, token or no authentication can specified.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+
+                <xs:attribute name="username" type="xs:string" >
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                Optional username (will be sent as basicauth).
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+
+                <xs:attribute name="password" type="xs:string" >
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                Optional password (will be sent as basicauth).
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+
+                <xs:attribute name="token" type="xs:string" >
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                Optional token.
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:assert vc:minVersion="1.1"
+                           test="(@username and @password and not(@token)) or (not(@username) and not(@password) and @token) or (not(@username) and not(@password) and not(@token))"
+                           xerces:message="Username/password must be specified or a token must be specified."
+                           saxon:message="Username/password must be specified or a token must be specified."
```

Comment:
The third case of none being specified is not in the message.

Reply:
Fair point. Since this assertion is just preventing the error case, how do you feel about this message instead?:

> Username/password and a token cannot both be specified.

I actually kind of want to replace the assertion test with a conditional that is a bit easier for me to grok, but since that might just be me, I'd like to get your opinion on that:
```
<xs:assert vc:minVersion="1.1"
                           test="if (@username or @password) then @username and @password and not(@token)
                                 else if (@token) then not(@username) and not(@password)
                                 else true()"
                           xerces:message="Username/password and a token cannot both be specified."
                           saxon:message="Username/password and a token cannot both be specified."/>
```

The main difference is in the approach taken. In the current assertion, all valid states are enumerated. In the proposed assertion, only exclusionary states are enumerated with a default pass for all other states. In practice, that only really allows us to replace the no-auth state with the catch-all state, so no big difference there. It's just something about the structure that makes things clearer in my mind. Thoughts?